source/common/stats/statsd.cc

```diff
@@ -33,44 +33,56 @@ Writer::~Writer() {
   }
 }
 
-void Writer::writeCounter(const std::string& name, uint64_t increment) {
-  std::string message(fmt::format("envoy.{}:{}|c", name, increment));
-  send(message);
-}
-
-void Writer::writeGauge(const std::string& name, uint64_t value) {
-  std::string message(fmt::format("envoy.{}:{}|g", name, value));
-  send(message);
-}
-
-void Writer::writeTimer(const std::string& name, const std::chrono::milliseconds& ms) {
-  std::string message(fmt::format("envoy.{}:{}|ms", name, ms.count()));
-  send(message);
-}
-
-void Writer::send(const std::string& message) {
+void Writer::write(const std::string& message) {
   ::send(fd_, message.c_str(), message.size(), MSG_DONTWAIT);
 }
 
 UdpStatsdSink::UdpStatsdSink(ThreadLocal::SlotAllocator& tls,
-                             Network::Address::InstanceConstSharedPtr address)
-    : tls_(tls.allocateSlot()), server_address_(address) {
+                             Network::Address::InstanceConstSharedPtr address, const bool use_tag)
+    : tls_(tls.allocateSlot()), server_address_(address), use_tag_(use_tag) {
   tls_->set([this](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
     return std::make_shared<Writer>(this->server_address_);
   });
 }
 
 void UdpStatsdSink::flushCounter(const Counter& counter, uint64_t delta) {
-  tls_->getTyped<Writer>().writeCounter(counter.name(), delta);
+  std::string message(
```

Comment:
nit: const here and similar places, or just kill local variable and do inline as var not needed.

Reply:
Yep, use const d3fb9ef450971c1a177af1644d2355f981e46ff8.